### PR TITLE
feat(generation): added options for token generation;

### PR DESCRIPTION
**Pull request by florinrinja**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -175,42 +185,47 @@ function parseToken (jwt, callback, now = Date.now()) {
...
 
-  if (_header.alg !== ALGORITHM_NAME) {
-    return callback(new Error('Algorithm not accepted for JSON Web Token. Only ' + ALGORITHM_NAME + ' is accepted'));
+  if (!ALLOWED_ALGORITHMS.includes(_header.alg)) {
+    return callback(new Error(`Algorithm not accepted for JSON Web Token. Only ${ALLOWED_ALGORITHMS.join(', ')} are accepted`));
   }
 
   if (_payload.exp && now > parseInt(_payload.exp, 10) * 1000) {
-    return callback(new Error('JSON Web Token expired'), _payload);
+    console.log('parse',{_payload});
```
Comment (matthieurobin) [inline index.js:193]: `console.log` to remove

File: index.js [hunk 2]
```diff
@@ -175,42 +185,47 @@ function parseToken (jwt, callback, now = Date.now()) {
...
  * @param  {[type]}   tokenString token base64 string
  * @param  {[type]}   signature   signature only
  * @param  {[type]}   publicKey   public key
  * @param  {Function} callback(err, parsedPayload)
  */
-function verifyToken (payload, tokenString, signature, publicKey, callback) {
+function verifyToken (header, payload, tokenString, signature, publicKey, callback) {
   try {
-    let _isValidSignature = ecdsa.verify(tokenString, signature, publicKey);
+    console.log({header});
```
Comment (matthieurobin) [inline index.js:217]: to remove

File: index.js [hunk 3]
```diff
@@ -240,14 +256,23 @@ function verify (jwt, publicKey, callback, now = Date.now()) {
  * @param  {Object} data     user data
  */
 function getToken (clientId, serverId, privKey, data) {
```
Comment (matthieurobin) [inline index.js:258]: `getToken` should have the same parameters as `generate`?

I mean, getToken is using generate function internaly, so we must have the same parameters (especially the options) to allow algorithm selection and so on.

Comment (florinrinja) [reply index.js:258]: yeah, okay. 
it's what i thought also.
i left it like this to limit the breaking changes 🙃

Comment (matthieurobin) [reply index.js:258]: If you don't impact this function with new parameters, we can potentially renew a token with a different algorithm.

File: package.json [hunk 4]
```diff
@@ -23,7 +23,10 @@
     "jsonwebtoken": "9.0.0",
     "mocha": "10.2.0",
     "should": "13.2.3",
-    "timekeeper": "2.2.0"
+    "timekeeper": "2.2.0",
+    "eslint": "8.44.0",
+    "eslint-plugin-json": "^3",
+    "eslint-plugin-vue": "^9"
```
[the remark this answers is not in the document]
Comment (florinrinja) [reply package.json]: removed `"eslint-plugin-vue": "^9"`
`"eslint-plugin-json": "^3",` needed for `eslint --fix`

File: index.js (hunk 3 above)
Comment (florinrinja) [reply index.js]: done, m'sieur

File: README.md [hunk 5]
```diff
@@ -122,16 +122,24 @@ can be used for another web-service which have the same clientId and public key.
...
   Re-use this same token during about 12 hours if called more than twice<br>
   Generate a new token automatically before expiration (20-minute before) or if privKey change
-
-  - clientId  : JWT issuer, token.iss
-  - serverId  : JWT audience, token.aud
-  - privKey   : private key
-
+  - options : 
+    - header : {
+      $~~$alg,  $~~$:$~~$  algorithm name
```
Comment (matthieurobin) [inline README.md:132]: what are those $?

Comment (matthieurobin) [reply README.md:132]: the rendering is change : 
image

Comment (florinrinja) [reply README.md]: done => changed rendering

**Pull request merged**